psydac/feec/tests/test_pushforward.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+import pytest
+
+from psydac.api.discretization  import discretize
+from sympde.topology            import ScalarFunctionSpace
+from sympde.topology            import Square
+from sympde.topology            import Mapping
+from psydac.mapping.discrete    import SplineMapping
+from psydac.feec.pushforward    import Pushforward
+
+def test_basic_call():
+
+    logical_domain = Square('Omega')
+
+    ncells = [2**4, 2**4]
+    degree = [2, 2]
+
+    # Mapping and physical domain
+    class CollelaMapping2D(Mapping):
+
+        _ldim = 2
+        _pdim = 2
+        _expressions = {'x': 'a * (x1 + eps / (2*pi) * sin(2*pi*x1) * sin(2*pi*x2))',
+                        'y': 'b * (x2 + eps / (2*pi) * sin(2*pi*x1) * sin(2*pi*x2))'}
+
+    mapping = CollelaMapping2D('M', a=1, b=1, eps=.2)
+    domain  = mapping(logical_domain)
+
+    hat_V0 = ScalarFunctionSpace('hat V0', logical_domain)
+    domain_h = discretize(logical_domain, ncells = ncells, periodic = [False, True])
+    hat_V0_h = discretize(hat_V0, domain_h, degree = degree)
+    
+    grid_x1 = hat_V0_h.breaks[0]
+    grid_x2 = hat_V0_h.breaks[1]
+
+    F = SplineMapping.from_mapping(hat_V0_h, mapping.get_callable_mapping())    
+    Pushforward(grid=(grid_x1, grid_x2), mapping=F, grid_type=0)
+
+    F = mapping   
+    Pushforward(grid=(grid_x1, grid_x2), mapping=F, grid_type=0)
```

Comment:
Unit tests should have an `assert` statement. What are you testing here exactly?

Maybe it is time to merge the modules `feec.pushforward` (which does not have any dedicated unit tests) and `feec.pull_push` (which has lots of tests). We should probably create a dedicated issue about that.

Reply:
the original code would raise a type error somewhere, so no assert was needed to make it fail...

Reply:
what kind of assert would you use, if we are just checking that the code runs ?

Reply:
In general I would check that the `Pushforward` class does what it is intended to do, and have some cases where the code was crashing before this PR.

Reply:
The real issue is that we have no proper unit tests for this class, as `test_postprocessing.py` is not extensive enough. The same appears to be true also for `pull_push.py`. I am creating an issue for this.

Reply:
See issue #395

Reply:
you are absolutely right, but here I just wanted to fix one bug :)